deploy/deploy.go

```diff
@@ -233,7 +284,36 @@ func (k *KindSpec) checkDependencies() error {
 			errs.Add(fmt.Errorf("install dependency %q to deploy", bin))
 		}
 	}
-	return errs.Err()
+	if errs.Err() != nil {
+		return errs.Err()
+	}
+	if k.Version != "" {
+		wantV, err := k.getKindVersion(k.Version)
+		if err != nil {
+			return err
+		}
+
+		if err := vExec.Exec("kind", "version"); err != nil {
+			return fmt.Errorf("failed to get kind version: %w", err)
+		}
+		// Reset buffer for next read
+		defer vOut.Reset()
+
+		vKindFields := strings.Fields(vOut.String())
+		if len(vKindFields) < 2 {
+			return fmt.Errorf("failed to parse kind version from: %s", vOut)
+		}
+
+		gotV, err := k.getKindVersion(vKindFields[1])
+		if err != nil {
+			return fmt.Errorf("kind version check failed: %w", err)
+		}
+		if gotV.Less(wantV) {
+			return fmt.Errorf("kind version check failed: got %s, want %s", gotV, wantV)
+		}
+		log.Infof("kind version valid: got %s want %s", gotV, wantV)
```

Comment:
above (line 112) we check the existence of docker and kubectl, we should also check those versions (and likely need to check against the Image field)

Reply:
this is just step one